app/models/mechanic.rb

```diff
@@ -0,0 +1,10 @@
+class Mechanic < ApplicationRecord
+  has_many :mechanic_rides
+  has_many :rides, through: :mechanic_rides
+
+
+  def add_ride(ride_id)
+    ride = Ride.find(ride_id)
+    self.rides << ride
+  end
```

Comment:
This model method is ok to be here, but like I mentioned before, you can actually do this work in the controller as well.